### ISSUE TITLE
AzureDevOps Build - YAML AWS Connection bug ?

### ISSUE THREAD

**Issue by steven-cuthill-otm** [aws/aws-vsts-tools#109]
Hello, 

Believe I have found a bug in the way the YAML build works over the Visual Designer when setting the awsCredentials property to the displayName of the AWS Service Connection i get the following error during build. you can see my YAML build sample below. I suspect that it's expecting the UUID for the connection rather than the name. I have given the connection liberal access rights so i don't think its down to ACL

Job Job1: Step  input awsCredentials references service connection Stevens TEMP which could not be found. The service connection does not exist or has not been authorized for use. For authorization details, refer to https://aka.ms/yamlauthz.

```
pool:
  vmImage: 'Ubuntu 16.04'

steps:

- task: AmazonWebServices.aws-vsts-tools.AWSPowerShellModuleScript.AWSPowerShellModuleScript@1
  displayName: 'CloudFormation Valadation Test'
  inputs:
    awsCredentials: 'Stevens TEMP'

    regionName: 'eu-west-1'

    filePath: 'build/cf-testScript.ps1'
```

**Comment by steven-cuthill-otm** [comment]
just bumping this as its a little bit of an issue as its broken our yaml build jobs

**Comment by steveataws** [comment]
We'll try and schedule some engineer time to take a look but we have next to no knowledge of what 'yaml builds' are right now, nor any idea why or current endpoint definition should not 'just work'. 

I wonder if we need to update the version of the vsts-task-lib we build against (currently 2.0.6), but if so we'd need to ensure backwards compatibility with Team Foundation Server 2017.

While we come up to speed on this apsect and do some investigation could you help us out with a step-by-step on how to construct a yaml build? What makes it different from what went before?

**Comment by steven-cuthill-otm** [comment]
Hello,

This link will help a lot I think getting you guys up to speed.
https://docs.microsoft.com/en-us/azure/devops/release-notes/2017/nov-28-vsts#configuration-as-code-yaml-builds-in-public-preview-build-tagimgrelease-notes-tagbuildpng

Its basically Build pipeline as code and not using the visual designer  where each task is defined as a YAML object. Build file is saved in your report and the build pipeline will run that file.  You can export the current task YAML definition by clicking on the View YAML link in the top right hand side.

I think the issue is how we are dealing with the awsCredentials, if I export the build pipeline (its done in JSON) I can see the input looks this and the awsCredentials is in a UUID format and not the display name.

I am going to do some more testing to see if I can make it work by using the UUID from the exported json.


                        "inputs": {
                            "awsCredentials": "b088b6ec-xxxx-xxxx-xxxxx-4ec6f5651f69",
                            "regionName": "eu-west-1",
                            "arguments": "",
                            "scriptType": "filePath",
                            "filePath": "build/cf-testScript.ps1",
                            "errorActionPreference": "stop",
                            "failOnStderr": "false",
                            "ignoreLASTEXITCODE": "false",
                            "workingDirectory": ""
                        }



[cid:image001.png@01D45A73.D9B05D60]
[cid:image002.png@01D45A73.D9B05D60]

Steven Cuthill
DevOps Manager

**Comment by steveataws** [comment]
Thanks for the info, I'll check it out. 

While looking around myself, I came across [this docs page](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=vsts&tabs=yaml%2Cbatch) that talks about a System.AccessToken variable which I don't see in your first example. Since endpoints are credentials, it made me wonder if perhaps a different mechanism is used for yaml builds and instead of referring to the endpoint by name (or uuid), maybe it's provided to the yaml build as a token by some other process that your build then needs to 'import'?

**Comment by ChastinaLi** [comment]
Just want to help you out here, I ran into the same issue. It turns out that if you didn't include a credentials parameter in the yaml file when you first created the build, it will not work even if you added the credentials param later on by updating the yaml file.

So, the solution is to delete the build from the VSTS webpage, and re-create it again (WITH the AWSCredentials parameter in your YAML already). And it works like a charm.

I found this from another thread https://github.com/Microsoft/azure-pipelines-agent/issues/1809#issuecomment-420222694.
